### PR TITLE
Add assembly var in variables pane

### LINKED ISSUE #53: Add feature to view assembly instructions in the values pane in the IDE
- Obtain assembly instructions from GDB corresponding to a stack frame
- Populate a variable called "instructions" with a String value that contains a snippet of instructions

### PR DESCRIPTION
**Pull request by jchen10500**
Add `$asm` as an additional slot in the variables pane. 
Screen Shot 2022-07-22 at 09 10 47

Fixes #53

### CODE REVIEW

File: src/main/java/jdwp/JDWP.java [hunk 1]
```diff
@@ -52,6 +53,8 @@ public class JDWP {
     static Map<Integer, MIFrame> framesById = new HashMap<>();
 
     static Map<Integer, LocalVariableImpl> localsByID = new HashMap<>();
+    static ArrayList<LocalVariableImpl> locals = new ArrayList<>();
+    static ReferenceTypeImpl strRef;
```
Comment (aavarghese) [inline src/main/java/jdwp/JDWP.java:57]: What is this `strRef` for?

File: src/main/java/jdwp/JDWP.java [hunk 2]
```diff
@@ -52,6 +53,8 @@ public class JDWP {
     static Map<Integer, MIFrame> framesById = new HashMap<>();
 
     static Map<Integer, LocalVariableImpl> localsByID = new HashMap<>();
+    static ArrayList<LocalVariableImpl> locals = new ArrayList<>();
```
Comment (aavarghese) [inline src/main/java/jdwp/JDWP.java:56]: Can we leave comments on why these static variables are needed?

File: src/main/java/jdwp/JDWPMethod.java [hunk 3]
```diff
@@ -177,12 +177,44 @@ public void reply(GDBControl gc, PacketStream answer, PacketStream command) {
...
+//                    answer.writeInt(method.argSlotCount());
+//                    answer.writeInt(variables.size());
+//                    for (LocalVariableImpl variable : variables) {
+//                        SlotInfo.write(variable, gc, answer);
+//                    }
+//
+//                } catch (AbsentInformationException e) {
+//                    answer.pkt.errorCode = JDWP.Error.ABSENT_INFORMATION;
+//                }
+//            }
```
Comment (aavarghese) [inline src/main/java/jdwp/JDWPMethod.java:217]: Can we remove these lines?

File: src/main/java/jdwp/JDWPMethod.java [hunk 4]
```diff
@@ -247,6 +279,17 @@ public void reply(GDBControl gc, PacketStream answer, PacketStream command) {
                         }
                     }
 
+                    // Add assmebly variable
```
Comment (aavarghese) [inline src/main/java/jdwp/JDWPMethod.java:282]: typo

File: src/main/java/jdwp/JDWPStackFrame.java [hunk 5]
```diff
@@ -115,6 +118,32 @@ public void reply(GDBControl gc, PacketStream answer, PacketStream command) {
                                     answer.writeNullObjectRef();
                             }
                         }
+                    } else if (vmVar.name().equals("$asm")) { // To do: create a new StringReferenceImpl
+                        ObjectReferenceImpl strObject = JDWP.strRef.instances(10000).get(0);
+                        StringReferenceImpl newStrRef = new StringReferenceImpl(strObject.referenceType(), null, "myStr");
+                        JDWP.strRef.instances(1).add(newStrRef);
+                        answer.writeByte(tag);
+//                        answer.writeObjectRef(newStrRef.uniqueID());
```
Comment (aavarghese) [inline src/main/java/jdwp/JDWPStackFrame.java:126]: if we aren't using the string ref stuff yet, can we remove all that from the PR? You should leave them in your branch for experiments till we get it to work.

**Comment by jchen10500** [comment]
In Fib's `result()` :
Screen Shot 2022-07-22 at 13 54 38
Screen Shot 2022-07-22 at 13 54 44

**Pull request merged**